### DISCUSSION TITLE
Dreame D10s Pro and Co - AI camera capatibilies

### DISCUSSION THREAD

**Discussion by Clusters** [Hypfer/Valetudo#1723] [Q&A / Support] [answered]
Hi Hypfer,
first of all thanks for your great job supporting Cloud-less robot functionality!

I have a very specific question about the new AI camera Dreame models. They are currently not supported (for anyone who needs more information continue reading here https://github.com/Hypfer/Valetudo/discussions/1690#discussioncomment-4710074), but possibly they might be in the future. My question is about if that ever happens:
Will there be potentially a possibility to get access to the live video feed on the robot?

Inside the repo it seems to be a still in progress topic. ;-)
`case MIOT_SERVICES.VACUUM_2.PROPERTIES.AI_CAMERA_SETTINGS.PIID:
      //ignored for now`

Why I'm asking?
I currently considering buying the D10s Pro over the L10 Pro, even if it is quite a bet, because of the rooting. 
The only big difference between those two seems to be the camera. One which is rumoured to avoid obstacles worse for the newer D10s Pro. So why buying it, if you don't want to use the cloud and therefore the live video feed feature? At least, if it is not possible to access the live feed without the cloud. ;-)

**Comment by Hypfer** [comment, marked as answer]
From a technical standpoint, the answer to your question is "most likely no" as the way dreame implemented that camera streaming feature likely won't be uncloudable. It works very different from everything else we do.

From a project standpoint, the answer to your question is "most definitely no" due to a few reasons ordered by priority

- IP Cameras are scary technology with many _many_ **many** options to be abused
	- Think for example: Abusive Tech-Partner in a relationship with other members of the household that have far less understanding of the technology used. It would be remarkably easy to covertly stalk them. I will not enable that.
	- The above is just one example. There are many more.
	- The best way to prevent abuse of some tool is by not having it in the first place
- Security is *hard*. Even more so when the stakes are high
	- Currently, if someone gains access to your robot, the worst they can do is play the locate sound at 3am and see your floor plan
	- With real-time video (and maybe audio) streaming, there are much much much worse things an attacker could do to you
- Real-time video processing is
	- complicated
	- not doable with the current Valetudo architecture
	- resource-intensive and thus risks overall stability of the robot, breaking core operation
- There are much better options to have live video feed 
	- Just buy a 30€ IPCAM
	- IPCams exist and have so for a long time
	- IPCams work. It's their job to provide a video feed
	- IPCams can be placed somewhere where it makes sense to place a camera. They're usually not mounted at foot-height.
	- IPCams look like IP cams, communicating to everyone in the household that it is a camera
- It's a gimmick
	- It's a functionality that will be used a few times because "hey, neat! a feature" and then be forgotten
- It's overall just a lot of work

**Reply by Clusters** [reply]
Thanks for the very detailed answer @Hypfer !
I was afraid that would be the answer. So I will stick with the L10 Pro option then :)

One tiny extra question. The D10s Pro's camera has some local avoidance detection implementation, which does not relay on a cloud connection, right? Otherwise it would have only Lidar for navigation.

**Reply by Hypfer** [reply]
The AI Camera obstacle avoidance does work just fine with Valetudo.
The ML model used to detect objects is run locally on the robot so it pretty much behaves just like another sensor with it using a camera + ML object recognition being a mere implementation detail
